forms/OpenIDPluginSettingsForm.php

```diff
@@ -0,0 +1,237 @@
+     *
+     * @return array complete list of enabled provider including all necessary endpoint URL's
+     */
+    private function _createProviderList(?array $providerList, ?array $providerListDB): array
+    {
+        $providerListResult = [];
+        if (isset($providerList) && is_array($providerList)) {
+            foreach ($providerList as $name => &$provider) { // Note: Use reference to modify $provider directly
+                if (!($provider['active'] ?? false)) {
+                    continue;
+                }
+
+                $providerDB = $providerListDB[$name] ?? null;
+                if (is_array($providerListDB) && $providerDB) {
+                    // Simplified checks and assignments for clientId and clientSecret
+                    $provider['clientId'] = $provider['clientId'] ?? '';
+                    if (empty($provider['clientId']) || $provider['clientId'] == self::HIDDEN_CHARS) {
+                        $provider['clientId'] = !empty($providerDB['clientId']) ? $providerDB['clientId'] : '';
+                    }
+
+                    $provider['clientSecret'] = $provider['clientSecret'] ?? '';
+                    if (empty($provider['clientSecret']) || $provider['clientSecret'] == self::HIDDEN_CHARS) {
+                        $provider['clientSecret'] = !empty($providerDB['clientSecret']) ? $providerDB['clientSecret'] : '';
+                    }
+                }
+
+                $openIdConfig = $this->_loadOpenIdConfig($provider['configUrl'] ?? '');
+                if (is_array($openIdConfig)) {
+                    $provider['authUrl'] = $openIdConfig['authorization_endpoint'] ?? null;
+                    $provider['tokenUrl'] = $openIdConfig['token_endpoint'] ?? null;
+                    $provider['userInfoUrl'] = $openIdConfig['userinfo_endpoint'] ?? null;
+                    $provider['certUrl'] = $openIdConfig['jwks_uri'] ?? null;
+                    $provider['logoutUrl'] = $openIdConfig['end_session_endpoint'] ?? null;
+                    $provider['revokeUrl'] = $openIdConfig['revocation_endpoint'] ?? null;
+                    $providerListResult[$name] = $provider;
+                }
+            }
+            unset($provider); // Unset reference to avoid potential issues later
+        }
+
+        return $providerListResult;
+    }
+
+    /**
+     * Calls the .well-known/openid-configuration which is provided in the $configURL and returns the result on success
+     *
+     * @return mixed|null
+     */
+    private function _loadOpenIdConfig(string $configUrl)
+    {
+        $curl = curl_init();
+        curl_setopt_array(
+            $curl,
+            [
+                CURLOPT_URL => $configUrl,
+                CURLOPT_RETURNTRANSFER => true,
+                CURLOPT_HTTPHEADER => ['Accept: application/json'],
+                CURLOPT_POST => false,
+            ]
+        );
+        $result = curl_exec($curl);
```

Comment:
Should use `PKPApplication::getHttpClient` instead of the `curl` functions; that way the proxy configuration will be used, when it's given in `config.inc.php`.

Reply:
Changed that function to 

```
private function _loadOpenIdConfig(string $configUrl)
    {
        $headers = [
            'Accept' => 'application/json',
        ];

        $httpClient = Application::get()->getHttpClient();
        try {
            $response = $httpClient->request(
                'GET',
                $configUrl,
                [
                    'headers' => $headers,
                    'allow_redirects' => ['strict' => true],
                ]
            );
        } catch (ClientException $exception) {
            return null;
        }

        return json_decode($response->getBody()->getContents(), true);
    }
```